src/main.rs

```diff
@@ -95,7 +99,11 @@ fn main() -> Result<()> {
                         return Some(input_file);
                     }
                 } else {
-                    println!("{formatted}");
+                    // When printing the result do not insert extra newlines.
+                    print!("{formatted}");
+                    if !formatted.ends_with('\n') {
```

Comment:
Why is this check needed? Wouldn't that imply that inplace mode could also produce files with a missing newline?

If possible, I'd drop that (which might make it easier to debug differences between inplace and direct output?

Reply:
Good point. In fact the formatter should already put a terminal newline for most inputs so this should add nothing.